chatsky/pipeline/pipeline/pipeline.py

```diff
@@ -43,66 +44,95 @@ class Pipeline(BaseModel, extra="forbid", arbitrary_types_allowed=True):
+    """
     script: Union[Script, Dict]
+    """
+    (required) A :py:class:`~.Script` instance (object or dict).
+    """
     start_label: NodeLabel2Type
+    """
+    (required) Actor start label.
+    """
     fallback_label: Optional[NodeLabel2Type] = None
+    """
+    Actor fallback label.
+    """
     label_priority: float = 1.0
+    """
+    Default priority value for all actor :py:const:`labels <dff.script.ConstLabel>`
+    where there is no priority. Defaults to `1.0`.
+    """
     condition_handler: Callable = Field(default=default_condition_handler)
+    """
+    Handler that processes a call of actor condition functions. Defaults to `None`.
+    """
     slots: GroupSlot = Field(default_factory=GroupSlot)
+    """
+    Slots configuration.
+    """
+    # Docs could look like this for one-liners
     handlers: Dict[ActorStage, List[Callable]] = Field(default_factory=dict)
+    """
+    This variable is responsible for the usage of external handlers on
+    the certain stages of work of :py:class:`~chatsky.script.Actor`.
+
+    - key: :py:class:`~chatsky.script.ActorStage` - Stage in which the handler is called.
+    - value: List[Callable] - The list of called handlers for each stage. Defaults to an empty `dict`.
+
+    """
     messenger_interface: MessengerInterface = Field(default_factory=CLIMessengerInterface)
+    """
+    An `AbsMessagingInterface` instance for this pipeline.
+    """
     context_storage: Union[DBContextStorage, Dict] = Field(default_factory=dict)
+    """
+    A :py:class:`~.DBContextStorage` instance for this pipeline or
+    a dict to store dialog :py:class:`~.Context`.
+    """
     before_handler: ComponentExtraHandler = Field(default_factory=list)
+    """
+    List of `_ComponentExtraHandler` to add to the group.
+    """
     after_handler: ComponentExtraHandler = Field(default_factory=list)
+    """
+    List of `_ComponentExtraHandler` to add to the group.
+    """
     timeout: Optional[float] = None
+    """
+    Timeout to add to pipeline root service group.
+    """
     optimization_warnings: bool = False
+    """
+    Asynchronous pipeline optimization check request flag;
+    warnings will be sent to logs. Additionally, it has some calculated fields:
+
+    - `_services_pipeline` is a pipeline root :py:class:`~.ServiceGroup` object,
+    - `actor` is a pipeline actor, found among services.
+
+    """
     parallelize_processing: bool = False
-    actor: Optional[Actor] = None
-    _services_pipeline: Optional[ServiceGroup] = None
+    """
+    This flag determines whether or not the functions
+    defined in the ``PRE_RESPONSE_PROCESSING`` and ``PRE_TRANSITIONS_PROCESSING`` sections
+    of the script should be parallelized over respective groups.
+    """
     _clean_turn_cache: Optional[bool]
 
-    def _create_actor(self) -> Actor:
+    @computed_field
+    @cached_property
+    def actor(self) -> Actor:
```

Comment:
Could you add tests that these properties are not changed if any field of pipeline is changed?
```python
actor = id(pipeline.actor)
pipeline.start_label = other
assert actor == id(pipeline.actor)
```